diff --git a/DataFormats/Provenance/interface/Hash.h b/DataFormats/Provenance/interface/Hash.h
index f1b52f2..60a0447 100644
--- a/DataFormats/Provenance/interface/Hash.h
+++ b/DataFormats/Provenance/interface/Hash.h
@@ -3,6 +3,7 @@
 
 #include <string>
 #include <functional>
+#include "DataFormats/Provenance/interface/HashedTypes.h"
 
 /*----------------------------------------------------------------------
   
@@ -211,5 +212,6 @@ namespace edm {
     return h.print(os);
   }
 
+typedef Hash<ModuleDescriptionType> ModuleDescriptionID;
 }  // namespace edm
 #endif
diff --git a/DataFormats/Provenance/interface/HashedTypes.h b/DataFormats/Provenance/interface/HashedTypes.h
index d85e024..5d1e3c9 100644
--- a/DataFormats/Provenance/interface/HashedTypes.h
+++ b/DataFormats/Provenance/interface/HashedTypes.h
@@ -14,5 +14,4 @@ namespace edm {
     ParentageType
   };
 }
-
 #endif  // DataFormats_Provenance_HashedTypes_h
diff --git a/DataFormats/Provenance/src/classes.h b/DataFormats/Provenance/src/classes.h
index 540983d..f1b5f54 100644
--- a/DataFormats/Provenance/src/classes.h
+++ b/DataFormats/Provenance/src/classes.h
@@ -56,6 +56,6 @@
 #include "DataFormats/Provenance/interface/RunAux.h"
 #include "DataFormats/Provenance/interface/RunLumiEntryInfo.h"
 
-namespace edm {
-  typedef Hash<ModuleDescriptionType> ModuleDescriptionID;
-}
+//namespace edm {
+//  typedef Hash<ModuleDescriptionType> ModuleDescriptionID;
+//}
